myndla-api/src/main/scala/no/ndla/myndlaapi/service/ArenaReadService.scala

```diff
@@ -482,6 +484,37 @@ trait ArenaReadService {
       } yield apiTopic
     }
 
+    def upvotePost(postId: Long, user: MyNDLAUser)(session: DBSession = AutoSession): Try[api.Post] = {
+      for {
+        maybePost     <- arenaRepository.getPost(postId)(session)
+        (post, owner) <- maybePost.toTry(NotFoundException(s"Could not find post with id $postId"))
+        upvoted       <- arenaRepository.getUpvoted(postId, user.id)(session)
+        _ <-
+          if (upvoted.isEmpty && owner.exists(_.id != user.id)) arenaRepository.upvotePost(postId, user.id)(session)
+          else Success(())
+        flags   <- arenaRepository.getFlagsForPost(postId)(session)
+        upvotes <- arenaRepository.getUpvotesForPost(postId)(session)
+        // The post is now upvoted if it wasn't initiated by the post's owner
+        newUpvotedState = upvoted.isEmpty && !owner.exists(_.id != user.id)
+        compiledPost    = CompiledPost(post, owner, flags, upvotes.length, newUpvotedState)
+        replies         = getRepliesForPost(compiledPost.post.id, user)(session).?
+      } yield converterService.toApiPost(compiledPost, user, replies)
+    }
+
+    def unUpvotePost(postId: Long, user: MyNDLAUser)(session: DBSession = AutoSession): Try[api.Post] = {
+      for {
+        maybePost     <- arenaRepository.getPost(postId)(session)
+        (post, owner) <- maybePost.toTry(NotFoundException(s"Could not find post with id $postId"))
+        upvoted       <- arenaRepository.getUpvoted(postId, user.id)(session)
+        _             <- if (upvoted.isDefined) arenaRepository.unUpvotePost(postId, user.id)(session) else Success(())
+        flags         <- arenaRepository.getFlagsForPost(postId)(session)
+        upvotes       <- arenaRepository.getUpvotesForPost(postId)(session)
+        newUpvotedState = !upvoted.isDefined // The upvote is now removed
+        compiledPost    = CompiledPost(post, owner, flags, upvotes.length, newUpvotedState)
```

Comment:
Resultatet her er uansett false.
```suggestion
        compiledPost    = CompiledPost(post, owner, flags, upvotes.length, upvoted = false)
```